[MTIA] Support bfloat16 in half_to_float

patrick-toulme opened the pull request:
Summary: Support bfloat16 in half_to_float when tracing a softmax op that has half_to_float=True.

Test Plan: CI

Differential Revision: D87607452

patrick-toulme commented:
@pytorchbot label "topic: not user facing

patrick-toulme commented on torch/_meta_registrations.py:
```diff
@@ -8101,7 +8101,7 @@ def meta_scaled_grouped_mm(
 def softmax(x: Tensor, dim: int, half_to_float: bool) -> Tensor:
     if half_to_float:
-        assert x.dtype == torch.half
+        assert x.dtype in [torch.half, torch.bfloat16]
```
@zou3519  test added and all checks are green

zou3519 reviewed (approved):
looks good but I do have comments, please address them before merging

patrick-toulme commented:
@pytorchbot merge

patrick-toulme commented:
@pytorchbot merge

patrick-toulme commented:
@pytorchbot merge

patrick-toulme committed:
[MTIA] Support bfloat16 in half_to_float (#168938)

Summary:

Support bfloat16 in half_to_float when tracing a softmax op that has half_to_float=True.

Test Plan: CI

Reviewed By: zou3519, marvin-kim

Differential Revision: D87607452

patrick-toulme commented:
@pytorchbot merge
